test/jdk/java/awt/Window/WindowIconUpdateOnDPIChanging/WindowIconUpdateOnDPIChangingTest.java

```diff
@@ -0,0 +1,137 @@
+            + "  <li>Check that the window icon is updated, it should display \"24\" on a screen"
+            + "      with a 150% DPI scaling, or \"32\" on a screen with a 200% DPI scaling."
+            + "      <br>If so, press <b>PASS</b>, otherwise press <b>FAIL</b>.</li>"
+            + "<ol>\n"
+            + "</body>\n"
+            + "</html>\n";
+
+    public static void main(String[] args) throws Exception {
+        countDownLatch = new CountDownLatch(1);
+        SwingUtilities.invokeLater(WindowIconUpdateOnDPIChangingTest::createUI);
+        countDownLatch.await(15, TimeUnit.MINUTES);
+        if (!testResult) {
+            throw new RuntimeException("Test fails!");
+        }
+    }
+
+    private static void createUI() {
+        frame = new JFrame("Window Icon Update Test");
+        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        frame.setSize(640, 480);
+        frame.addWindowListener(new WindowAdapter() {
+            public void windowClosed(WindowEvent e) {
+                countDownLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new BorderLayout());
+        frame.getContentPane().add(createInstrumentsPane(), BorderLayout.CENTER);
+        frame.getContentPane().add(createControlPanel(), BorderLayout.SOUTH);
+        frame.setIconImages(IntStream.rangeClosed(16, 32).mapToObj(WindowIconUpdateOnDPIChangingTest::createIcon).toList());
+        frame.setVisible(true);
+    }
+
+    private static JTextPane createInstrumentsPane() {
+        JTextPane instructionsPane = new JTextPane();
+        instructionsPane.setContentType("text/html");
+        instructionsPane.setText(INSTRUCTIONS);
+        instructionsPane.setEditable(false);
+        instructionsPane.setBorder(new EmptyBorder(10, 10, 10, 10));
+        return instructionsPane;
+    }
+
+    private static JPanel createControlPanel() {
+        JButton passButton = new JButton("Pass");
+        passButton.addActionListener(e -> testResult = true);
+        passButton.addActionListener(e -> frame.dispose());
+
+        JButton failButton = new JButton("Fail");
+        failButton.addActionListener(e -> frame.dispose());
+
+        JPanel controlPanel = new JPanel(new FlowLayout(FlowLayout.CENTER));
+        controlPanel.add(passButton);
+        controlPanel.add(failButton);
+
+        return controlPanel;
+    }
+
+    /**
+     * Creates an icon of the size specified. The size is drawn at the center of the icon.
+     *
+     * @param size the size of the icon
+     */
+    private static Image createIcon(int size) {
+        BufferedImage image = new BufferedImage(size, size, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D g = image.createGraphics();
+        g.setFont(new Font("dialog", Font.BOLD, 12));
+        g.setColor(Color.BLACK);
+        g.drawString(String.valueOf(size), 0, size - (size - g.getFont().getSize()) / 2);
```

Comment:
Before drawing black text, I suggest filling the background with white. I will guarantee the icon can always been seen. On the Windows taskbar which is black, it's very hard to see the black text. In the dark theme, where the window title bar is black, it'll be impossible to see the difference.